app/models/lead_provider_delivery_partnership.rb

```diff
@@ -19,4 +19,15 @@ class LeadProviderDeliveryPartnership < ApplicationRecord
 
   scope :with_delivery_partner, ->(delivery_partner_id) { where(delivery_partner_id:) }
   scope :with_active_lead_provider, ->(active_lead_provider_id) { where(active_lead_provider_id:) }
+  scope :for_contract_period, ->(contract_period) {
+    joins(:active_lead_provider)
+      .where(active_lead_providers: { contract_period_year: contract_period.year })
+      .includes(active_lead_provider: :lead_provider)
+  }
+  scope :active_lead_provider_ids_for, ->(delivery_partner, contract_period) {
+    where(delivery_partner:)
+      .joins(:active_lead_provider)
+      .where(active_lead_providers: { contract_period_year: contract_period.year })
+      .select(:active_lead_provider_id)
+  }
```

Comment:
I think this should be a service rather than a scope - I always think of scopes as things that filter a recordset but don't change the shape. That can go in as a housekeeping issue to tidy up though, no need for it to be a stopper.